src/components/App/index.js

```diff
@@ -0,0 +1,111 @@
+import { actions } from '../ReduxAppWrapper/store/actions/'
+import { validJWT } from '~utils/jwt'
+
+class ModalApp extends Component {
+  constructor(props) {
+    super(props)
+    this.events = new EventEmitter()
+    this.events.on('complete', this.trackOnComplete)
+    Tracker.setUp()
+    Tracker.install()
+    this.bindEvents(props.options.onComplete, props.options.onError)
+  }
+
+  componentDidMount() {
+    this.prepareInitialStore({}, this.props.options)
+  }
+
+  componentDidUpdate(prevProps) {
+    this.jwtValidation(prevProps.options, this.props.options)
+    this.prepareInitialStore(prevProps.options, this.props.options)
+    this.rebindEvents(prevProps.options, this.props.options);
+  }
+
+  componentWillUnmount() {
+    this.props.socket && this.props.socket.close()
+    this.events.removeAllListeners('complete', 'error')
+    Tracker.uninstall()
+  }
+
+  jwtValidation = (prevOptions = {}, newOptions = {}) => {
+    if (prevOptions.token !== newOptions.token) {
+      if (!validJWT(newOptions.token)) {
+        this.onInvalidJWT()
+      }
+    }
+  }
+
+  onInvalidJWT = () => {
+    const type = 'exception'
+    const message = 'Invalid token'
+    this.events.emit('error', { type, message })
+  }
+
+  trackOnComplete = () => Tracker.sendEvent('completed flow')
+
+  bindEvents = (onComplete, onError) => {
+    this.events.on('complete', onComplete)
+    this.events.on('error', onError)
+  }
+
+  rebindEvents = (oldOptions, newOptions) => {
+    this.events.off('complete', oldOptions.onComplete)
+    this.events.off('error', oldOptions.onError)
+    this.bindEvents(newOptions.onComplete, newOptions.onError)
+  }
+
+  prepareInitialStore = (prevOptions = {}, options = {}) => {
+    const { userDetails: { smsNumber } = {}, steps } = options
+    const { userDetails: { smsNumber: prevSmsNumber } = {}, steps: prevSteps } = prevOptions
+
+    if (smsNumber && smsNumber !== prevSmsNumber) {
+      this.props.actions.setMobileNumber(smsNumber)
+    }
+
+    if (steps && steps !== prevSteps) {
+      const enabledDocs = enabledDocuments(steps)
+      if (enabledDocs.length === 1) {
+        this.props.actions.setIdDocumentType(enabledDocs[0])
+      }
+    }
+  }
+
+  render = ({ options: { useModal, isModalOpen, onModalRequestClose, containerId, shouldCloseOnOverlayClick, ...otherOptions }, ...otherProps }) => {
+    return (
+      <LocaleProvider language={this.props.options.language}>
+        <Modal useModal={useModal} isOpen={isModalOpen} onRequestClose={onModalRequestClose} containerId={containerId} shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}>
+          <Router options={{ ...otherOptions, events: this.events }} {...otherProps} />
+        </Modal>
+      </LocaleProvider>
+    )
+  }
+}
+
+
+const mapStateToProps = (state) => ({
+  ...state.globals,
+  captures: state.captures
+})
+
+const mapDispatchToProps = (dispatch) => ({
+  actions: bindActionCreators(actions, dispatch)
```

Comment:
You use only two actions: setMobileNumber & setIdDocumentType. I would prefer to pass only them:

```
const mapDispatchToProps = {
  setMobileNumber: actions.setMobileNumber,
  setIdDocumentType: actions.setIdDocumentType,
}
```

If you wish these be scoped under "actions", then maybe something like:

```
const mapDispatchToProps = dispatch => ({
  actions: {
    setMobileNumber: () => dispatch(actions.setMobileNumber()),
    setIdDocumentType: () => dispatch(actions.setIdDocumentType()),
  },
});
```

Reply:
If I do that I will have to pass actions to every single component and it starts to get complex when you use a lot of HoC and composition. Also, this is consistent with the original implementation. In this PR we just want to make the redux store available only within the SDK, so I will leave this task for a different story. Thanks for the suggestion though.

Reply:
Sure, no problem.

I mentioned some time earlier, that purpose of connecting to Redux is that you can connect many times instead of prop-drilling.